mir_eval/util.py

```diff
@@ -616,9 +664,8 @@ def match_events(ref, est, window):
         ``matching[i] == (i, j)`` where ``ref[i]`` matches ``est[j]``.
 
     """
-
     # Compute the indices of feasible pairings
-    hits = np.where(np.abs(np.subtract.outer(ref, est)) <= window)
+    hits = np.where(outer_distance(ref, est) <= window)
```

Comment:
Does it make more sense to parameterize this by function, rather than letting the user simply supply their own hits array?  Alternately, is the "outer" distance the level of abstraction you want, or would it be easier to allow a user to supply a point-wise dissimilarity function?

As it was originally written, this function did two things:

1. Find feasible matchings (hits) between ref and est under a specific definition of hit
2. Convert hits to a graph and find an optimal matching.

Maybe a better factorization of this functionality might be to just implement 2, and have a separate function for 1.  Then we can provide a few different hit-matrix constructors that all feed into the same matching alg.

I don't have strong opinions on this though, so maybe it's overkill to factor it that far.